### DISCUSSION TITLE
Error while downloading Soil data

### DISCUSSION THREAD

**Discussion by ParameshwariS-tiger** [APSIMInitiative/ApsimX#8825] [Q&A] [answered]
Hi all, I was getting an error related to soil temperature while trying to run Peanut module with the downloaded soil data from apsim

Below is the image of downloaded soil component for a particular location and the corresponding reporting variables:
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/456a6de7-8f99-4521-96f5-4d3dc3c25fe6)
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/9238b78e-3408-422b-8c7d-55f2648a1de5)

And the error was:
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/e5991ab4-9533-46b0-be5c-b2b66c4b30ce)

The image of default soil component and its reporting variable:
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/c0e4b18d-e909-4250-af9a-a32d311c5a74)
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/ca3f5e8b-8c08-49ed-b552-d54decb58ce2)

However, I replaced the "CERESSoilTemperature" with "Temperature" similar to the default soil component (mentioned above) which resolved the problem.

1. Is the solution correct? If not, how to resolve it?
2. Why is the temperature component changing for downloaded soil? And what is the difference between "CERESSoilTemperature" and "Temperature"?

Any assistance on this would be greatly appreciated, thanks:)

**Comment by hol353** [comment]
Yes, what you have done is correct. The aflotoxin calculation in the peanut model doesn't work with the default CERESSoilTemperature model. It needs the newer SoilTemperature model.

**Reply by ParameshwariS-tiger** [reply]
Thanks @hol353 for you answer, however I have tried removing _[Peanut].AflotoxinRisk_ from the report variables but still getting the below error:
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/6ba1a63a-1921-4c83-825d-ba85486cfa2f)

Any insights on this would be really helpful, thanks:)

**Reply by ric394** [reply]
@par456 this may be an edge case for the recent locator changes

**Comment by ric394** [comment, marked as answer]
@ParameshwariS-tiger To get this work without error, replace CERESSoilTemperature's or SoilTemperature's name with 'Temperature' and make sure you rename your downloaded soil to 'Soil'.

**Reply by ParameshwariS-tiger** [reply]
@ric394 Yes, I've done the same to address the issue. I just wanted to double-check if that approach is correct. Thank you for your response! It works now:)
